chat-service/internal/app/repositories/mongodb/chat_repository.go

```diff
@@ -116,6 +133,12 @@ func (r *ChatRepository) Get(ctx context.Context, userID string) ([]*models.Chat
 }
 
 func (r *ChatRepository) SendMessage(ctx context.Context, chatID models.ChatID, authorID, content string) (*models.Message, error) {
+	if deadline, ok := ctx.Deadline(); !ok || time.Until(deadline) > r.config.Timeout {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, r.config.Timeout)
+		defer cancel()
+	}
+
```

Comment:
Вариант 1: выносишь эту логику в отдельную функцию и уже ее вызываешь
```suggestion
         ctx, cancel := WithRepositoryTimeout(ctx, r.config.Timeout)
	defer cancel()
```
Вариант 2 (тоже самое но чуть сложнее):

```go
func WithRepositoryContext[T any](ctx context.Context, timeout time.Duration, fn func(ctx context.Context) (T, error)) (T, error) {
	if deadline, ok := ctx.Deadline(); !ok || time.Until(deadline) > timeout {
		ctx, cancel := context.WithTimeout(ctx, timeout)
		defer cancel()
		return fn(ctx)
	}
	return fn(ctx)
}
```
```
func (r *ChatRepository) SendMessage(ctx context.Context, chatID models.ChatID, authorID, content string) (*models.Message, error) {
	return WithRepositoryContext[*models.Message](ctx, r.config.Timeout, func(ctx context.Context) (*models.Message, error) {
		return r.doSendMessage(ctx, chatID, authorID, content)
	})
}
```